contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,428 @@
+
+/**
+ * <p>
+ * AvroToPojo
+ * </p>
+ * A generic implementation for conversion from Avro to POJO. The POJO class
+ * name & field mapping should be provided by the user. If this mapping is not
+ * provided then reflection is used to determine this mapping. As of now only
+ * primitive types are supported.
+ *
+ * @displayName Avro To Pojo
+ * @category Converter
+ * @tags avro
+ * @since 3.3.0
+ */
+
+@InterfaceStability.Evolving
+public class AvroToPojo implements Operator
+{
+
+  private List<Field> columnNames;
+
+  private Class<?> cls;
+
+  private static final String FIELD_SEPARATOR = ":";
+  private static final String RECORD_SEPARATOR = ",";
+
+  private String genericRecordToPOJOFieldsMapping = null;
+
+  public String getGenericRecordToPOJOFieldsMapping()
+  {
+    return genericRecordToPOJOFieldsMapping;
+  }
+
+  public void setGenericRecordToPOJOFieldsMapping(String genericRecordToPOJOFieldsMapping)
+  {
+    this.genericRecordToPOJOFieldsMapping = genericRecordToPOJOFieldsMapping;
+  }
+
+  private String schemaFile;
+
+  private Schema schema;
+
+  private List<FieldInfo> fieldInfos;
+
+  private transient List<ActiveFieldInfo> columnFieldSetters = null;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  @AutoMetric
+  int fieldErrorCnt = 0;
+
+  public void parseSchema() throws IOException
+  {
+    setSchema(new Schema.Parser().parse(getSchemaString()));
+  }
+
+  public String getSchemaString()
+  {
+    return schemaFile;
+  }
+
+  public void setSchemaString(String schemaFile)
+  {
+    this.schemaFile = schemaFile;
+  }
+
+  public Schema getSchema()
+  {
+    return schema;
+  }
+
+  public void setSchema(Schema schema)
+  {
+    this.schema = schema;
+  }
+
+  public byte[] serialize(Object obj) throws IOException
+  {
+    try (ByteArrayOutputStream b = new ByteArrayOutputStream()) {
+      try (ObjectOutputStream o = new ObjectOutputStream(b)) {
+        o.writeObject(obj);
+      }
+      return b.toByteArray();
+    }
+  }
+
+  @InputPortFieldAnnotation(optional = false)
+  public final transient DefaultInputPort<GenericRecord> data = new DefaultInputPort<GenericRecord>()
+  {
+
+    @Override
+    public void process(GenericRecord tuple)
+    {
+
+      try {
+        Object obj = getPOJOFromGenericRecord(tuple, getCls());
+
+        if (obj != null) {
+          output.emit(obj);
+          recordCnt++;
+        }
+
+      } catch (InstantiationException | IllegalAccessException e) {
+        LOG.error("Could not initialize object of class - " + getClass().getName());
+        errorCnt++;
+      }
+    }
+
+  };
+
+  @SuppressWarnings("unchecked")
+  public Object getPOJOFromGenericRecord(GenericRecord tuple, Class<?> cls)
+      throws InstantiationException, IllegalAccessException
+  {
+    Object newObj = getCls().newInstance();
+
+    try {
+
+      for (int i = 0; i < columnFieldSetters.size(); i++) {
+
+        AvroToPojo.ActiveFieldInfo afi = columnFieldSetters.get(i);
```

Comment:
Can you rename the variables to have meaningful names e.g. afi can be fieldInfo or activeFieldInfo?